src/Tests/Eshopworld.Messaging.Tests/MessengerTopicTest.cs

```diff
@@ -79,14 +92,64 @@ public async Task Test_ReceiveCreatesTheTopic()
             ServiceBusFixture.ServiceBusNamespace.AssertSingleTopicSubscriptionExists(typeof(TestMessage), subscriptionName);
         }
     }
 
-    [EswEventName("test.prova.nino")]
-    class TestNinoEvent : DomainEvent
+    [Fact, IsLayer1]
+    public async Task Test_ReceiveWithTopicName_CreatesTheTopic()
+    {
+        await ServiceBusFixture.ServiceBusNamespace.ScorchNamespace();
+        var topicName = nameof(Test_ReceiveWithTopicName_CreatesTheTopic).Replace("_", "");
+        topicName = $"{topicName}_Topic";
+
+        using (IDoFullMessaging msn = new Messenger(ServiceBusFixture.ConfigSettings.ServiceBusConnectionString, ServiceBusFixture.ConfigSettings.AzureSubscriptionId))
+        {
+            var subscriptionName = nameof(Test_ReceiveWithTopicName_CreatesTheTopic).Replace("_", "");
+            await msn.Subscribe<TestMessage>(_ => { }, subscriptionName, topicName);
+            msn.CancelReceive<TestMessage>(topicName);
+            ServiceBusFixture.ServiceBusNamespace.AssertSingleTopicSubscriptionExists(topicName, subscriptionName);
+        }
+    }
+
+    [Theory, IsLayer1]
+    [InlineData(-2)]
+    [InlineData(0)]
+    [InlineData(4)]
+    public async Task Test_ReceiveWithTopicNameAndIdleTime_WhenLessThan5Min_ThrowsException(int minutes)
     {
+        await ServiceBusFixture.ServiceBusNamespace.ScorchNamespace();
+        
+        var topicName = nameof(Test_ReceiveWithTopicNameAndIdleTime_WhenLessThan5Min_ThrowsException).Replace("_", "");
 
+        using (IDoFullMessaging msn = new Messenger(ServiceBusFixture.ConfigSettings.ServiceBusConnectionString, ServiceBusFixture.ConfigSettings.AzureSubscriptionId))
+        {
+            var subscriptionName = $"IdleTopicSubscription{minutes}";
+            Func<Task> action = async () => await msn.Subscribe<TestMessage>(_ => { }, subscriptionName, topicName,
+                deleteOnIdleDurationInMinutes: minutes);
+            action.Should().Throw<ArgumentException>().Which.ParamName.Should()
+                .BeEquivalentTo("deleteOnIdleDurationInMinutes");
+        }
+    }
+
+    [Fact, IsLayer1]
+    public async Task Test_ReceiveWithTopicNameAndIdleTime_CreatesTheTopicAndRemovesAfterIdle()
+    {
+        await ServiceBusFixture.ServiceBusNamespace.ScorchNamespace();
+        
+        const string topicName = "IdleTopicTest";
+        const int idleTimeMin = 5;
+
+        using (IDoFullMessaging msn = new Messenger(ServiceBusFixture.ConfigSettings.ServiceBusConnectionString, ServiceBusFixture.ConfigSettings.AzureSubscriptionId))
+        {
+            const string subscriptionName = "IdleTopicSubscription";
+            await msn.Subscribe<TestMessage>(_ => { }, subscriptionName, topicName, deleteOnIdleDurationInMinutes: idleTimeMin);
+            msn.CancelReceive<TestMessage>(topicName);
+            ServiceBusFixture.ServiceBusNamespace.AssertSingleTopicSubscriptionExists(topicName, subscriptionName);
+            await Task.Delay(TimeSpan.FromMinutes(6));

```

Comment:
We should not have such delays in our tests but I can see this pattern is used in many of our tests. We may need to introduce a separate test category like "LongRunning" to separate such tests from other unit tests.
For now, we may at least reduce it by some seconds to save time :) as it needs to be just over 5 mins here.

Reply:
I tested locally with 5m10s and 5m20s and it didn't work. So I think we can stick to the 6 min. I agree it is not ideal. But 5 min is the min you can set for autodelete idle subscriptions.